nixos/modules/services/networking/dnscrypt-proxy2.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }: with lib;
+
+let
+  cfg = config.services.dnscrypt-proxy2;
+
+  json = pkgs.writeText "dnscrypt-proxy.json" (builtins.toJSON cfg.config);
+
+  toml = pkgs.runCommand "dnscrypt-proxy.toml" {} ''
+    ${pkgs.remarshal}/bin/json2toml < ${json} > $out
+  '';
+in
+
+{
+  options.services.dnscrypt-proxy2 = {
+    enable = mkEnableOption "dnscrypt-proxy2";
+
+    config = mkOption {
```

Comment:
This option implements the approach described in NixOS/rfcs#42, and as such it should be renamed to `settings` instead.